app.js

```diff
@@ -13,10 +13,34 @@ Project.prototype.toHtml = function(){
   $('#portfolio').append(theCompiledHtml);
 }
 
-projectData.forEach(function(item){
-  var newProject = new Project(item);
-  newProject.toHtml();
-});
+function buildProjects(data){
+  data.forEach(function(item){
+    var newProject = new Project(item);
+    newProject.toHtml();
+  });
+}
+
+function initProjects(){
+  var eTag;
+  $.ajax({
+    type: 'HEAD',
+    url: 'data/projects.json',
+    complete: function(XMLHttpRequest){
+      eTag = XMLHttpRequest.getResponseHeader('ETag');
```

Comment:
To avoid confusion with the type name (`XMLHttpRequest`), it would be better to name this with the camelCase variable name convention (typically `xhr` or `jqXHR`).